hw/ip/entropy_src/doc/_index.md

```diff
@@ -10,83 +10,143 @@ This module conforms to the [Comportable guideline for peripheral functionality.
+- Optional Vendor Specific tests, which allow silicon creators to extend the health checks by adding a top-level block external to this IP.
+
+The Repetition Count and Adaptive Proportion test are specifically recommended by SP 800-90B, and are implemented in accordance with those recommendations.
+In FIPS/CC-compliance mode, these latter three checks are performed on fixed window of data, typically consisting of 1024 bits each.
+Per the definition in SP 800-90B, the Repetition Count test does not operate on a fixed window.
+The repetition count test fails if any sequence of bits continuously asserts the same value for too many samples, as determined by the programmable threshold, regardless of whether that sequence crosses any window boundaries.
+The thresholds for these tests should be chosen to achieve a low false-positive rate (&alpha;) given a conservative estimate of the manufacturing tolerances of the physical noise source.
+The combined choice of threshold and window size then determine the false-negative rate (&beta;), or the probability of missing statistical defects at any particular magnitude.
+
+In order to compensate for the fact our tests (like *all* realistic statistical tests) have finite resolution for detecting defects, we conservatively use 1024 bits of raw digital noise to construct each 384 bit key.
+When passed through the conditioning block, the resultant entropy stream will be full entropy unless the physical noise source has encountered some statistical defect serious enough to reduce the raw min-entropy to a level below 0.375 bits of entropy per output bit.
+We choose this level as our definition of "non-tolerable statistical defects" for the purposes of evaluating this system under AIS31.
+Given this definition of "non-tolerable defects", the health-checks as implemented for this block will almost certainly detect any of the previously mentioned defects in a single iteration of the health checks (i.e. such serious defects will be detected with very low &beta;).
+
+In addition to the brief, low-latency health checks, various long-term statistics are accumulated in registers for additional diagnostic purposes or for in-depth analysis.
+For the hardware interface, entropy bits will be transferred whenever the downstream consumer is ready to receive them.
 
-At a high-level, this ENTROPY_SRC block, when enabled, will continuously collect entropy bits from the entropy source into a FIFO that can be read from the TLUL bus, or sent out through a hardware interface.
-The rng source will generate a relative slow set of bit streams, in the tens of kilohertz range.
+At a high-level, this ENTROPY_SRC block, when enabled, will continuously collect entropy bits from the entropy source into a FIFO that can be read from the TL-UL bus, or sent out through a hardware interface.
```

Comment:
Offline comment from @zi-v:
Are we still with 'continuously' I thought we add an option for periodic or on demand.

Reply:
Hi @zi-v:
The periodic is not yet implemented but is listed in future features.
The on-demand can be implemented today, simply by disabling the entropy source when not needed. (Check out my other comment near line 149)